pkg/rbac/userData.go

```diff
@@ -307,7 +307,9 @@ func (user *UserDataCache) getSSRRforNamespace(ctx context.Context, cache *Cache
 							// No need to loop through all resources. Save the wildcard *
 							// exit the resourceRulesLoop
 							if res == "*" && api == "*" {
+								// user.nsrCache.lock.Lock()
```

Comment:
Why are we removing this lock?
I think it's still needed because it's possible to have multiple requests from the same user. Maybe we could simplify by locking at a higher level.

Reply:
Adding that causes the rbac.(*UserDataCache).getNamespacedResources function to timeout and fail in Test_getNamespaces_emptyCache